Applications/Android/NNDetector/app/src/main/jni/simpleshot.cpp

```diff
@@ -0,0 +1,444 @@
+bool use_detection_for_test_prototypes = false;
+
+std::vector<std::string> split(std::string input, char delimiter) {
+  std::vector<std::string> answer;
+  std::stringstream ss(input);
+  std::string temp;
+  while (getline(ss, temp, delimiter)) {
+    answer.push_back(temp);
+  }
+  return answer;
+}
+
+int trainData_cb(float **input, float **label, bool *last, void *user_data) {
+  auto data =
+    reinterpret_cast<nntrainer::simpleshot::DirDataLoader *>(user_data);
+  data->next(input, label, last);
+  return 0;
+}
+
+ml::train::Model *initialize_det(int argc, char *argv[]) {
+
+  if (argc < 3) {
+    std::cerr << "usage: ./main [det backbone path] [det input image dim]\n";
+    throw std::invalid_argument("wrong argument");
+  }
+
+  std::string det_backbone_path = argv[1];
+  unsigned int det_input_image_dim = std::stoul(argv[2]);
+
+  ANDROID_LOG_D("---detection backbone_path: %s", det_backbone_path.c_str());
+  ANDROID_LOG_D("---input shape: %d:%d:3", det_input_image_dim,
+                det_input_image_dim);
+
+  std::string in_str = std::to_string(det_input_image_dim);
+
+  det_model = ml::train::createModel(ml::train::ModelType::NEURAL_NET,
+                                     {"batch_size=1", "epochs=1"});
+
+  LayerHandle backbone_layer = ml::train::layer::BackboneTFLite(
+    {"name=backbone", "model_path=" + det_backbone_path,
+     "input_shape=" + in_str + ":" + in_str + ":3", "trainable=false"});
+  det_model->addLayer(backbone_layer);
+
+  det_model->summarize(std::cout, ML_TRAIN_SUMMARY_MODEL);
+
+  std::shared_ptr<ml::train::Optimizer> optimizer;
+  optimizer = ml::train::optimizer::SGD({"learning_rate=0.1"});
+
+  if (det_model->setOptimizer(optimizer)) {
+    throw std::invalid_argument("failed to set optimizer");
+  }
+  if (det_model->compile()) {
+    throw std::invalid_argument("model compilation failed");
+  }
+  if (det_model->initialize()) {
+    throw std::invalid_argument("model initiation failed");
+  }
+
+  return det_model.get();
+}
+
+ml::train::Model *initialize_rec(int argc, char *argv[]) {
+
+  if (argc < 5) {
+    std::cerr << "usage: ./main [rec backbone path] "
+                 "[rec input image dim] [num class] [knn norm variant]\n";
+    throw std::invalid_argument("wrong argument");
+  }
+
+  std::string rec_backbone_path = argv[1];
+  unsigned int rec_input_img_dim = std::stoul(argv[2]);
+  unsigned int num_class = std::stoul(argv[3]);
+  std::string knn_norm_variant = argv[4];
+
+  ANDROID_LOG_D("---recognition backbone path: %s", rec_backbone_path.c_str());
+  ANDROID_LOG_D("---num classes: %d", num_class);
+  ANDROID_LOG_D("---input shape: %d:%d:3", rec_input_img_dim,
+                rec_input_img_dim);
+  ANDROID_LOG_D("---knn norm variant: %s", knn_norm_variant.c_str());
+
+  rec_model = ml::train::createModel(ml::train::ModelType::NEURAL_NET,
+                                     {"batch_size=1", "epochs=1"});
+
+  LayerHandle backbone_layer = ml::train::layer::BackboneTFLite(
```

Comment:
I think it is good example to use tflite backbone layer as an freezed subnetwork.